javascript/atoms/action.js

```diff
@@ -87,21 +87,25 @@ bot.action.clear = function(element) {
         'Element must be user-editable in order to clear it.');
   }
 
-  bot.action.LegacyDevice_.focusOnElement(element);
   if (element.value) {
+    bot.action.LegacyDevice_.focusOnElement(element);
     element.value = '';
     bot.events.fire(element, bot.events.EventType.CHANGE);
+    bot.events.fire(element, bot.events.EventType.BLUR);
+    bot.action.LegacyDevice_.focusOnElement(document.body);
   } else if (bot.dom.isElement(element, goog.dom.TagName.INPUT) &&
              (element.getAttribute('type') && element.getAttribute('type').toLowerCase() == "number")) {
     // number input fields that have invalid inputs
     // report their value as empty string with no way to tell if there is a
     // current value or not
+    bot.action.LegacyDevice_.focusOnElement(element);
     element.value = '';
```

Comment:
I'm not sure why we're not running the unfocussing steps here, or with the content editable, though both of these should (according to the spec) also be unfocussed.

Reply:
I'm not sure either. It's likely an oversight on my part. Though that branch doesn't fire the change event either (and never has).